File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.js
function solution(a, b, c) {
    const sum1 = a + b + c;
    const sum2 = a*a + b*b + c*c;
    const sum3 = a*a*a + b*b*b + c*c*c;
    
    const cnt = new Set([a, b, c]).size;
    return cnt === 3 ? sum1 : cnt === 2 ? sum1 * sum2 : sum1 * sum2 * sum3;
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.java
import java.util.*;

class Solution {
    public int solution(int a, int b, int c) {
        int sum1 = a + b + c;
        int sum2 = a*a + b*b + c*c;
        int sum3 = a*a*a + b*b*b + c*c*c;
        
        Set<Integer> set = new HashSet<>();
        set.add(a); set.add(b); set.add(c);
        
        int unique = set.size();
        if (unique == 3) return sum1;
        if (unique == 2) return sum1 * sum2;
        return sum1 * sum2 * sum3;
    }
}
